distributed-commandbus-springcloud/src/main/java/org/axonframework/springcloud/commandhandling/SpringHttpCommandBusConnector.java

```diff
@@ -181,7 +220,84 @@ public void onSuccess(CommandMessage<? extends C> commandMessage, R result) {
             super.complete(createReply(commandMessage, false, cause));
         }
-
     }
 
+    /**
+     * Builder class to instantiate a {@link SpringHttpCommandBusConnector}.
+     * The {@code localCommandBus} of type (@link CommandBus}, {@link RestOperations} and {@link Serializer} are
+     * <b>hard requirements</b> and as such should be provided.
+     */
+    public static class Builder {
+
+        private CommandBus localCommandBus;
+        private RestOperations restOperations;
+        private Serializer serializer;
+
+        /**
+         * Sets the {@code localCommandBus} of type {@link CommandBus} to publish received commands which to the local
+         * segment.
+         *
+         * @param localCommandBus the {@link CommandBus} to publish received commands which to the local segment
+         * @return the current Builder instance, for a fluent interfacing
+         */
+        public Builder localCommandBus(CommandBus localCommandBus) {
+            assertThat(localCommandBus,
+                       Objects::nonNull,
+                       () -> new AxonConfigurationException("LocalCommandBus may not be null"));
+            this.localCommandBus = localCommandBus;
+            return this;
+        }
+
+        /**
+         * Sets the {@link RestOperations} used to send commands to other nodes.
+         *
+         * @param restOperations the {@link RestOperations} used to send commands to other nodes
+         * @return the current Builder instance, for a fluent interfacing
+         */
+        public Builder restOperations(RestOperations restOperations) {
+            assertThat(restOperations,
+                       Objects::nonNull,
+                       () -> new AxonConfigurationException("RestOperations may not be null"));
+            this.restOperations = restOperations;
+            return this;
+        }
+
+        /**
+         * Sets the {@link Serializer} used to serialize command messages when they are sent between nodes.
+         *
+         * @param serializer the {@link Serializer} used to serialize command messages when they are sent between nodes
+         * @return the current Builder instance, for a fluent interfacing
+         */
+        public Builder serializer(Serializer serializer) {
+            assertThat(serializer,
+                       Objects::nonNull,
+                       () -> new AxonConfigurationException("Serializer may not be null"));
+            this.serializer = serializer;
+            return this;
+        }
+
+        /**
+         * Initializes a {@link SpringHttpCommandBusConnector} as specified through this Builder.
+         *
+         * @return a {@link SpringHttpCommandBusConnector} as specified through this Builder
+         */
+        public SpringHttpCommandBusConnector build() {
+            return new SpringHttpCommandBusConnector(this);
+        }
+
+        private void validate() {
+            assertThat(localCommandBus,
```

Comment:
Extraction to a separate method.

Reply:
Replaced for the `BuilderUtils` functions.
What do you think?